### DISCUSSION TITLE
does except(roomId) work with socket.id?

### DISCUSSION THREAD

**Discussion by SupertigerDev** [socketio/socket.io#4267] [Q&A] [answered]
I want to emit an event to everyone in the room but except 1 `socket.id`. will this work?

```js
io.in("room1").except(socket.id).emit("event", data)
```
in the [cheetsheat](https://socket.io/docs/v4/emit-cheatsheet/), it just says about rooms and not socket.id
![image](https://user-images.githubusercontent.com/8193841/152132964-fbcd8414-2b8d-41e0-9192-b8061699315e.png)

Thanks!

**Comment by darrachequesne** [comment, marked as answer]
Yes it does! I will update the documentation to make it clearer.
